pixelgl/input.go

```diff
@@ -384,26 +428,45 @@ func (w *Window) initInput() {
 			case glfw.Repeat:
 				w.tempInp.repeat[Button(key)] = true
 			}
+
+			for _, cbfun := range w.callbacks.buttonCallbacks {
+				go cbfun(Button(key), ButtonAction(action))
+			}
 		})
 
 		w.window.SetCursorEnterCallback(func(_ *glfw.Window, entered bool) {
 			w.cursorInsideWindow = entered
+
+			for _, cbfun := range w.callbacks.cursorEnterCallbacks {
+				go cbfun()
+			}
 		})
 
 		w.window.SetCursorPosCallback(func(_ *glfw.Window, x, y float64) {
 			w.tempInp.mouse = pixel.V(
 				x+w.bounds.Min.X,
 				(w.bounds.H()-y)+w.bounds.Min.Y,
 			)
+
+			for _, cbfun := range w.callbacks.cursorPosCallbacks {
+				go cbfun(x, y)
+			}
 		})
 
 		w.window.SetScrollCallback(func(_ *glfw.Window, xoff, yoff float64) {
 			w.tempInp.scroll.X += xoff
 			w.tempInp.scroll.Y += yoff
+
+			for _, cbfun := range w.callbacks.scrollCallbacks {
+				go cbfun(xoff, yoff)
+			}
 		})
 
 		w.window.SetCharCallback(func(_ *glfw.Window, r rune) {
 			w.tempInp.typed += string(r)
+			for _, cbfun := range w.callbacks.charCallbacks {
+				go cbfun(r)
```

Comment:
is there a reason you are doing this asynchronously with `go` instead of calling them sequentially? I feel like this could be an issue if callbacks overlap with each other. Not sure if that's a valid fear or not however. Would like to hearinput from you and @dusk125 on this...

Reply:
Say you register multiple callbacks for an event. Calling the callbacks asynchronously was done as to not keep one callback waiting for the previous one to finish. However, I see the reason for calling them sequentially, so that might be something that we could change.

Reply:
perhaps maybe a waitgroup then?

Reply:
My preference would be to call them sequentially, and leave it to the handler to do what it needs to async. Especially handling typing, things could get out of order which would be annoying.

Reply:
Maybe have async as a bool option somehow?